kbase/narrative
# DATAUP-703 job channel bandwidth

# Description of PR purpose/changes

It was noticed that the job channel of the websocket was using a fairly colossal amount of bandwidth. This wound up being due to all job status being returned from an `all_job_status` call in the kernel-side job monitoring thread, even if all of those jobs were complete and nothing on the front end was tracking them.

After some discussion, we decided that one of the cores of the problem was using the "refresh" flag for each job as an integer. This would represent the number of front end things (widgets, viewers, etc.) that wanted to receive continual job status. As those things were created or destroyed, the count would go up and down, and as long as one thing wanted job status, it would get returned on each turn of the cycle.

This is horribly confusing and inefficient and led to the tracking count being very brittle and prone to losing synchronicity. 

Instead, we decided to flip that to a boolean. If `True`, we return the latest job status on each cycle. This will work better as we're now in a position where there is effectively only one widget up at a time that will want job status. There's some internal developer debugging tools that also might want this, but they can be managed in other ways and aren't relevant for most, if not all, users.

An additional issue was uncovered where whenever a new job was created - whether by an app cell or by a direct code execution of `AppManager.run_app` - that job was set to always return its status. Thus, if you were to, say, write a short loop that generates 2000+ jobs, their status would always be returned, even long after all jobs were complete. The refresh value for them would also get set to 1 whenever the kernel restarted, or the page reloaded.

All of these together have the potential to gobble up all of a user's bandwidth pretty quickly, and might cause other problems in a narrative where there's 100s or 1000s of cells, due to network lag and such.

This PR, then, adds the following changes:
1. Turn that `refresh` int into a boolean.
2. On job creation, leave `refresh=False` until requested by a front end element (already built in to App Cells)
3. On kernel start or restart, if given a list of cell ids, turn `refresh=True` for those jobs that are incomplete AND associated with a cell.

# Jira Ticket / Issue #

Related Jira ticket: https://kbase-jira.atlassian.net/browse/DATAUP-703

# Testing Instructions
* Details for how to test the PR:
* Effectively there should be no visual changes
  * starting up new single app cells should work as expected
    * newly run app cells should get data until they stop running
    * running a new app cell, then refreshing the page should still get job status
    * running a new app cell, then restarting the kernel, should still get job status
  * all of the above should apply to bulk import cells as well

# Dev Checklist:


# Updating Version and Release Notes (if applicable)
